### DISCUSSION TITLE
Radxa Rock 5B (Huge IO improvement over Pi4)

### DISCUSSION THREAD

**Discussion by Stefankante** [outdoorbits/little-backup-box#199] [Q&A]
I just bought Rock 5B bc of its sheer power and IO performance wish to just install some Debian linux which I tried.
- DietPi OS 
- Armbian Bullseye 
- Full Debian desktop default to Rock 5B

![IMG_3726](https://user-images.githubusercontent.com/35141284/229886907-f55bc086-4c32-44b0-bb3c-ce4e7a4834a6.jpg)

only DietPi manage to get it running and can access menu through Web UI. It mounted drive either target or source but seems to don't know the file system and failed to transfer. Maybe this is bc of differences in File structure compare to Raspian OS? 

Maybe I should tried to install DietPi on Pi4 and tried to install LBB first to see how it perform. 
But if any Rock 4 or 5 could use LBB it will open such huge opportunity to have super fast transfer speed as it perform something like 350MB/s on each drive independantly not on same USB bus. AND NVME!!!!

I will try DietPi on Pi4 first to see if it work. As it's only one distro which has multi-support across boards.
[https://dietpi.com](url)

**Comment by outdoorbits** [comment]
Did you read out the install-error.log generated during lbb's 
installation? Did it install all the packages? What filesystem types did 
you use on your USB?

**Reply by Stefankante** [reply]
same as before on Pi4 tested both Expat and HFS+ failed to mount. Let me reinstall it (I tried so many distro)

**Reply by Stefankante** [reply]
**GNU nano 5.4                    install-error.log**                             
  warnings.warn(
  running bdist_wheel
  running build
  running build_ext
  building 'spidev' extension
  creating build
  creating build/temp.linux-aarch64-cpython-39
  aarch64-linux-gnu-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g >
  error: command 'aarch64-linux-gnu-gcc' failed: No such file or directory
  ----------------------------------------
  ERROR: Failed building wheel for spidev
ERROR: Could not build wheels for spidev which use PEP 517 and cannot be instal>
no crontab for dietpi
no crontab for dietpi
debconf: delaying package configuration, since apt-utils is not installed
rm: cannot remove './tinyfilemanager': No such file or directory
Cloning into 'tinyfilemanager'...
Generating a RSA private key
................................+++++
.......................................................+++++
writing new private key to '/etc/ssl/private/apache-selfsigned.key'
-----
--2023-04-04 20:31:20--  https://davesteele.github.io/comitup/latest/davesteele>
Resolving davesteele.github.io (davesteele.github.io)... 2606:50c0:8001::153, 2>
Connecting to davesteele.github.io (davesteele.github.io)|2606:50c0:8001::153|:>
HTTP request sent, awaiting response... 200 OK
Length: 43052 (42K) [application/octet-stream]
Saving to: ‘davesteele-comitup-apt-source_latest.deb’

     0K .......... .......... .......... .......... ..        100% 1.39M=0.03s

2023-04-04 20:31:20 (1.39 MB/s) - ‘davesteele-comitup-apt-source_latest.deb’ sa>

Configuration file '/etc/apt/sources.list.d/davesteele-comitup.list', does not >
Installing new config file as you requested.
Failed to enable unit: Unit file NetworkManager.service does not exist.
debconf: delaying package configuration, since apt-utils is not installed
Unit dnsmasq.service does not exist, proceeding anyway.
Unit dhcpd.service does not exist, proceeding anyway.
Unit dhcpcd.service does not exist, proceeding anyway.
Unit wpa-supplicant.service does not exist, proceeding anyway.
sh: 1: cannot create /sys/class/leds/PWR/trigger: Directory nonexistent
sh: 1: cannot create /sys/class/leds/PWR/brightness: Directory nonexistent
fusermount: failed to unmount /media/iOS: Invalid argument
fusermount: failed to unmount /media/cloud: Invalid argument

**Reply by Stefankante** [reply]
HFS+ to HFS+ work just fine. but ExFat failed to both write and read, Any suggestion?

**Comment by outdoorbits** [comment]
Does it fail on mounting? What's the error message?

**Comment by outdoorbits** [comment]
There might me several problems about different packages in both 
distributions. For spi-support (some displays) it looks like a compiler 
is needed. In raspbian lbb enables i2c and spi for the display. This 
would also be a point to figure out.

I would hope, all the problems could be solved during installation. I 
don't have a rock pi, so I can't test this. But if you would like to 
play, I'll try to support you. If yes, please have a first look at 
install-little-backup-box.sh [1] line 142. You would have to check, if 
all that packages are available. Then check to error-log of the 
installation...

Links:
------
[1] 
https://github.com/outdoorbits/little-backup-box/blob/main/install-little-backup-box.sh

**Comment by Stefankante** [comment]
![Screen Shot 2566-04-05 at 04 05 39](https://user-images.githubusercontent.com/35141284/229923853-88b0a4c8-bfca-4bde-8270-4950f6fb7ee9.png)
So disappointed, It shows faster in synthetic speedtest reaching ~300MB/s per drive. then it maxed out like half again to roughly ~150MB/s (max) between drives.

It hovering around high 90+MB/s to ~120MB/s. in LBB with real footage video and photos. 
That's not worth the hassle compare to Pi4 which a bit slower but all done!!!! 

Jesus christ I need to sell this.

**Comment by outdoorbits** [comment]
But maybe, in a far future, when raspberry pi's are back in stock, there 
will be a RPi 5 ... ;-)

**Comment by Stefankante** [comment]
BTW Internal > USB (NVME 1500MB/s+ to USB SSD 300MB/s )
It takes quite considerably good bandwidth utilization. ~170-200MB/s
![Screen Shot 2566-04-05 at 23 30 31](https://user-images.githubusercontent.com/35141284/230145618-35bdd806-0149-40c9-8109-9bbea85de809.png)

I'm wondering if I bought USB3 or USB3.1,3.2 capable PCIe card (the one that has linux driver) and adapted to NVME slot. Will it boast the speed up. Seems like 5Gb/s Independent port speed on board was a lie lol.
